### ISSUE TITLE
Tailscale incompatible with KDE Connect

### ISSUE THREAD

**Issue by Gapoly** [tailscale/tailscale#14476]
### What is the issue?

KDE Connect can't connect to my computer if any of the two devices are connected to Tailscale

### Steps to reproduce

I activate Tailscale on either my smartphone or laptop and I lose the connection between with KDE Connect

### Are there any recent changes that introduced the issue?

No

### OS

Linux, Android

### OS version

Arch Linux, Android 14

### Tailscale version

1.78.1

### Other software

KDE Connect

**Comment by pelluch** [comment]
I've been having issues with the same. However, I have been able to get it to work under the following circumstances (Arch Linux, Android 15):

- The PC *not* using an exit node (it can be connected to Tailscale or not, but if it is, it won't work using an exit node)
- The Android device is connected to Tailscale, with or without an exit node enabled
- In the Tailscale Android app, Split tunneling is enabled and KDE connect is excluded from the tunnel
- In the Android VPN settings for Tailscale, "Block connections without VPN" must be off ("Always-on VPN" can be enabled).

While ideally I'd like to be able to toggle "Block connection without VPN", the above configuration has been an acceptable workaround for me.

**Comment by theMarix** [comment]
I also got it working with an exit node on the PC. Overall, you need to follow @pelluch 's instruction. The essential addition is:
* Enable “Allow LAN access” on the PC. This setting is by default disabled when using an exit node.

**Comment by izvyk** [comment]
Quite strange, for me it just works. As Tailscale brings two devices to the same LAN, they should find each other without any issues. I've just tested as follows:

• Laptop on WiFi
• Phone on mobile connection
• Both connected to Tailscale

They were able to find each other and pair with a request sent from PC to the phone and another way around. All the functions work as usual, just like if they were connected to the same LAN.

If it doesn't work you may try to add your device by the IP shown in the Tailscale admin console.

**Comment by L1ttleFlyyy** [comment]
There is [this post on Reddit](https://www.reddit.com/r/Tailscale/comments/w8japf/to_whom_it_may_concern_kde_connect_will_not_work/) and [portablesounds](https://www.reddit.com/user/portablesounds/) 's comment is very revealing:

> For anyone who stumbles on this post:
KDE Connect appears to rely on UDP broadcast packets. Unfortunately, tailscale does not support broadcast or multicast packets ([source](https://forum.tailscale.com/t/broadcast-multicast-support/156/2)).
Everyone else who got this to work must have already had both devices on the same subnet. I'm able to get it to work when I have "Allow LAN Access" set on my phone on the same WiFi, but if I turn that off or disconnect from my home network it breaks again.
TL;DR: You can have Tailscale on, you just have to have "Allow LAN Access" enabled on your phone and be on the same network.

**Comment by hyunbinseo** [comment]
I've got it working using this method. No other settings have been changed. Thank you @izvyk

> If it doesn't work you may try to add your device by the IP shown in the Tailscale admin console.

On the KDE Connect Android application:

1. (Hamburger Menu) Pair new device - ignore the searched device
2. (Kebab menu) Add devices by IP
3. (FAB, Plus button) Enter Tailscale IP of the PC
4. Go back and click the PC from the list - the pairing should work different networks (e.g. 5G on phone)

If this does not work, and if you can give up pairing on different networks, you could try excluding KDE from Tailscale Android application. (Settings / App split tunneling)

**Comment by Utopiah** [comment]
Does it work going THROUGH tailscale? By that I mean no side stepping tailscale via the LAN option but rather desktop at home with KDE Connect, tailscale on, then mobile on 4G (not LAN) and on same tailnet.

I tried today while walking back home, knowing my desktop was still on, and it didn't work.

**Comment by Delurkey** [comment]
You really just need to add the tailnet address to your IP list on the phone 😊

**Comment by Utopiah** [comment]
Damn, so obvious once you know it! It even works with device name : 

<img width="1080" height="2400" alt="Image" src="https://github.com/user-attachments/assets/499b887e-a6ea-49bd-a81d-1ae94f935780" />

<img width="1080" height="2400" alt="Image" src="https://github.com/user-attachments/assets/a28a6455-ea06-42cf-9e78-d20ae40297b6" />

Thanks @Delurkey !

**Comment by DimasInchidi** [comment]
-> from your terminal run `tailsacle ip`
-> from KDE Connect mobile app >> Pair new device >> Add devices by IP >> input the ipv4 (or) ipv6 from `tailsacle ip` output

i found this easier to setup than using bluetooth backend and feels more reliable actually, thanks everyone.

**Comment by gabfv** [comment]
The above steps didn't seem to work much for me under iOS with KDE Connect but somehow that worked with "Disable UDP Broadcast" enabled under advanced settings in the KDE iOS app. 

If I enable UDP Broadcast again, KDE can't see my computer through tailscale, oddly enough.

**Comment by Valkryst** [comment]
Confirming that using "Disable UPD Broadcast" causes it to work under iOS.

**Comment by rmcmilli** [comment]
I'm on Android/Fedora. I previously had this working by setting the tailscale IP manually but it broke recently. I was only able to get it to pair by adding KDEconnect to the split tunnel list then removing it.

**Comment by greggorievich** [comment]
> I'm on Android/Fedora. I previously had this working by setting the tailscale IP manually but it broke recently. I was only able to get it to pair by adding KDEconnect to the split tunnel list then removing it.

On Android/Pop!OS, I had this working up until very recently this way as well, using the hostname of my desktop in the Android app. I can see that enabling split tunneling for KDE Connect on my phone allows it to pair and connect, but if I remove it from the list I get disconnected pretty immediately (plus, with split tunneling if I'm not on my home WiFi I can't connect, which is if course why I use TailScale). 

I wonder what changed. 

EDIT: It appears that the KDE Connect app changed. I used APKmirror to install the previous version, 1.34.4, and am able to connect again via Tailscale.

**Comment by roberto-chang-silva** [comment]
Hi! Just wanted to share my experience in case it helps.

I actually was able to use KDE Connect over Tailscale in a kinda working way: from my phone app I could pair to my desktop by typing the Tailscale IP directly (100.x.x.x), and after pairing it works even remotely on mobile data (communication is bidirectional once paired). My desktop is CachyOS KDE, and I saw similar behavior with my laptop too (Fedora KDE latest).

But I couldn’t figure out how to do it the other way around (from desktop 2 phone). I wanted to learn the proper setup to add/pair the phone from the desktop side using the Tailscale IP, but it just didn’t work for me, so I left it as initiate from phone 2 desktop only.

Also: I still haven’t managed to get KDE Connect working via direct Tailscale IP between my laptop and my desktop (Linux from2 Linux). Phone from2 desktop works, but laptop from2 desktop doesn’t, so maybe it’s firewall/ports or discovery or something.

One more weird thing: when I was at university, KDE Connect showed a bunch of random devices in the available devices list (probably other people on the same LAN). So I disabled network discovery because I didn’t like being visible to everyone… and then I lost the connection over Tailscale too. When I enabled discovery again, it started working again.

So yeah: it would be awesome if KDE Connect could be limited to only Tailscale (only the tailscale interface / only Tailscale IPs) without LAN discovery making you visible on public networks.

If anyone has a better way to do this (especially laptop from2 desktop), I’d really appreciate it. Thanks!

**Comment by parnavh** [comment]
Yes after the new update 1.35.1, I'm able to ping the manual ip address but device doesn't show up. Rolling back to 1.34.4 worked!

**Comment by Zocker1999NET** [comment]
The recently introduced issue (with 1.35.1) is also tracked on [KDE‘s bugzilla as 515707](https://bugs.kde.org/show_bug.cgi?id=515707). TL;DR: They introduced a new security feature filtering for link-local IPs presumably preventing connectivity via VPNs & networks using 100.64.0.0/10 for their IPs, like Tailscale does (see [this commit](https://invent.kde.org/network/kdeconnect-android/-/commit/72881f8b04d280284b31ddd9810ee1c9a0fa8320)).

For this "general issue" I want to add that, my KDE Connect over Tailscale setup was working flawlessly. As already stated, that only required listing the Tailscale IPs/FQDNs on each KDE Connect client (so KDE Connect is able to "discover" the devices via Tailscale). On Android this is a setting called "Add devices by IP".

What is not noted here is that KDE Connect on "desktop" also has this setting, through it is not accessible by UI, but requires listing the IPs in a config file & restarting KDE Connect. The full explanation for this is provided on [KDE’s bugzilla on 475570, comment 1](https://bugs.kde.org/show_bug.cgi?id=475570#c1)

Setting that is required because when using exit nodes and/or routes overlapping with the local network, KDE Connect may not be able to connect to the other devices by itself. I assume the UDP discovery is failing in those cases. Setting "allow LAN access" in Tailscale on those devices may fix that for some setups, but not for mine.

**Comment by QazCetelic** [comment]
There is an open PR in the KDE Connect Android repository to resolve this issue https://invent.kde.org/network/kdeconnect-android/-/merge_requests/621

**EDIT:** https://invent.kde.org/network/kdeconnect-android/-/merge_requests/625 with the same fix has been merged instead. I think this can be closed now.

**Comment by roberto-chang-silva** [comment]
> There is an open PR in the KDE Connect Android repository to resolve this issue https://invent.kde.org/network/kdeconnect-android/-/merge_requests/621
> 
> **EDIT:** https://invent.kde.org/network/kdeconnect-android/-/merge_requests/625 with the same fix has been merged instead. I think this can be closed now.

Now kde connect through tailscale is working.

**Comment by queezle42** [comment]
I tried this using IPv6 tailnet IPs and that also doesn't work, due to https://github.com/tailscale/tailscale/issues/18799.

The problem is unrelated to this issue, but since KDE Connect shows the same error/behavior I thought I leave a link for anyone who finds this issue.

**Comment by croakerbcts** [comment]
Concur with https://github.com/tailscale/tailscale/issues/14476#issuecomment-3902328586 , this looks resolved. Closing.

**Comment by cagedbird043** [comment]
Adding one extra data point for people who find this issue later from Android + KDE Connect + Tailscale-like tailnet setups.

The 1.35.1 regression mentioned above appears to be the link-local / address filtering change in KDE Connect Android and was fixed on the KDE side by the linked MR. However, I hit a second, similar-looking failure mode while testing KDE Connect Android over a Tailscale-compatible tailnet endpoint provided by `sing-box` on Android.

In my case:

- the phone could already reach the desktop tailnet IP (`100.64.x.x`);
- SSH over the tailnet worked;
- packet capture on the desktop showed TCP traffic reaching the phone on KDE Connect's port (`1716`);
- but KDE Connect Android still did not finish discovery/pairing.

The important observation was that the Android app's Java `ServerSocket` did not see the remote peer as `100.64.x.x`. The accepted socket showed up as loopback instead, for example:

```text
127.0.0.1:xxxxx -> 127.0.0.1:1716
```

So the packet was arriving through the tailnet, but by the time it was delivered to KDE Connect Android, it looked like a local loopback connection. KDE Connect's LAN backend then rejected it before reading the identity packet, because it was neither a trusted LAN address nor the manually configured custom-device address.

This is probably not the exact same bug as the 1.35.1 `100.64.0.0/10` filtering regression, but the user-facing symptom is very similar: "manual Tailscale IP is reachable, but KDE Connect still does not pair".

If anyone is debugging this class of problem, I would suggest checking both layers:

1. Does `Add devices by IP` contain the peer's Tailscale/tailnet IP or MagicDNS name?
2. Does the Android app actually see the accepted TCP socket as the peer IP, or as `127.0.0.1` because of a userspace VPN/tailnet/proxy path?

A safe application-side fix should not blindly trust all loopback traffic. The approach that worked in my test fork was to treat loopback inbound as an explicit userspace-tailnet/proxy entry path only when the user has configured custom devices. In other words: if the user opted into explicit peers, do not reject the loopback-delivered connection before reading the KDE Connect identity packet.

Hopefully this helps future people distinguish:

- broadcast/multicast discovery not working over Tailscale;
- the fixed KDE Connect Android 1.35.1 address-filtering regression;
- and userspace Android VPN/tailnet implementations that deliver inbound peer TCP to the app as loopback.
